coordinator/src/main/scala/filodb.coordinator/queryplanner/ShardKeyRegexPlanner.scala

```diff
@@ -35,12 +47,18 @@ class ShardKeyRegexPlanner(dataset: Dataset,
     }
   }
 
-  private def getNonMetricShardKeyFilters(logicalPlan: LogicalPlan) = LogicalPlan.
+  private def getNonMetricShardKeyFilters(logicalPlan: LogicalPlan): Seq[Seq[ColumnFilter]] = LogicalPlan.
     getRawSeriesFilters(logicalPlan).map { s => s.filter(f => dataset.options.nonMetricShardColumns.contains(f.column))}
 
   private def generateExec(logicalPlan: LogicalPlan, nonMetricShardKeyFilters: Seq[ColumnFilter],
-                           qContext: QueryContext) = shardKeyMatcher(nonMetricShardKeyFilters).map(f =>
-                           queryPlanner.materialize(logicalPlan.replaceFilters(f), qContext))
+                           qContext: QueryContext): Seq[ExecPlan] = {
+    val queryParams = qContext.origQueryParams.asInstanceOf[PromQlQueryParams]
+    shardKeyMatcher(nonMetricShardKeyFilters, queryParams.promQl).map(f => {
+      val newQueryParams = queryParams.copy(promQl = f.updatedQuery)
+      val newQueryContext = qContext.copy(origQueryParams = newQueryParams)
+      queryPlanner.materialize(logicalPlan.replaceFilters(f.updatedFilters), newQueryContext)
+    })
```

Comment:
Formatting:

```
something.map( f => {
...
})
```
change to 

```
something.map { f =>
...
}
```

Please change in other code if applicable. Have seen you use the non-preferred formatting multiple times. :)

Reply:
Yes, Java style :)